tests/tasks/setup/eks/awscli-dp.yaml

```diff
@@ -0,0 +1,87 @@
+  steps:
+  - name: create-nodegroup
+    image: amazon/aws-cli
+    workingDir: $(workspaces.config.path)
+    script: |
+      ssh_key_name=$(params.region)-nodegroups-ssh-key
+      rm -f $HOME/.ssh/$SSH_KEY_NAME.pem
+      if [[ "$(aws ec2 --region $(params.region) describe-key-pairs --key-names $ssh_key_name --query 'KeyPairs[?KeyName==`'$ssh_key_name'`].KeyName' --output text)" -eq 0 ]]; then 
+        echo "KeyPair not found"; 
+        aws ec2 create-key-pair --region $(params.region) --key-name $ssh_key_name --query 'KeyMaterial' --output text > $HOME/.ssh/$SSH_KEY_NAME.pem
+      else 
+        echo "KeyPair already exists"; 
+      fi
+      
+      NG_SUBNETS=$(aws eks --endpoint $(params.endpoint) describe-cluster --name $(params.cluster-name) \
+      --query cluster.resourcesVpcConfig.subnetIds --output text \
+      --region $(params.region))
+
+      export NG_NAME=NG-$(params.cluster-name)-$(params.region)-$(date +%Y%m%d-%H%M%S)
+      
+      #create node group with SSH
+      aws eks --endpoint $(params.endpoint) create-nodegroup \
+      --cluster-name $(params.cluster-name) \
+      --nodegroup-name $NG_NAME \
+      --node-role $(params.host-cluster-node-role-arn) \
+      --region $(params.region) \
+      --remote-access ec2SshKey=$ssh_key_name \
+      --scaling-config minSize=$(params.min-nodes),maxSize=$(params.max-nodes),desiredSize=$(params.desired-nodes) \
+      --subnets $NG_SUBNETS
+
+      while [[ "$(aws eks --endpoint $(params.endpoint) --region $(params.region) describe-nodegroup --cluster-name $(params.cluster-name) --nodegroup-name $NG_NAME --query nodegroup.status --output text)" == "CREATING" ]]
+      do
+      echo "$NG_NAME is "CREATING" at $(date)"
+      sleep 10
+      done
+      aws eks --endpoint $(params.endpoint) update-kubeconfig --name $(params.cluster-name) --region $(params.region)
+      cp  /root/.kube/config $(workspaces.config.path)/kubeconfig
+  - name: validate-nodes
+    image: alpine/k8s:1.22.6
+    workingDir: $(workspaces.config.path)
+    script: |
+      mkdir -p /root/.kube/
+      cp $(workspaces.config.path)/kubeconfig /root/.kube/config
+      #kubectl commands are purely for knowing state of cluster before kicking off the test.
+      kubectl version
+      kubectl config current-context
+      kubectl describe clusterrole eks:node-manager
+      kubectl get nodes -o wide
+      kubectl get ns
+      while true; do
+          ready_node=$(kubectl get nodes 2>/dev/null | grep -w Ready | wc -l )
+          if [[ "$ready_node" -eq $(params.desired-nodes) ]]; then break; fi
+          sleep 10
```

Comment:
In the worst case we would wait for 10 seconds after all nodes are ready. Can we reduce it 2 seconds? I guess its alright to check once every couple of seconds